lib/daru/dataframe.rb

```diff
@@ -1720,46 +1853,19 @@ def access_vector_single_index *names
                                        index: @index, name: @name)
     end
 
-    def access_row *names
-      if @index.is_a?(MultiIndex)
-        access_row_multi_index(*names)
-      else
-        access_row_single_index(*names)
-      end
-    end
-
-    def access_row_multi_index *names
-      location = names.first
+    def access_row *indexes
+      positions = @index.pos(*indexes)
 
-      pos = @index[names]
-      if pos.is_a?(Integer)
-        return Daru::Vector.new(populate_row_for(pos), index: @vectors, name: pos)
-      end
-
-      new_rows = pos.map { |tuple| populate_row_for(tuple) }
-
-      if !location.is_a?(Range) && names.size < @index.width
-        pos = pos.drop_left_level names.size
-      end
-
-      Daru::DataFrame.rows(new_rows, order: @vectors, name: @name, index: pos)
-    end
-
-    def access_row_single_index *names
-      location = names.first
-
-      if names.count < 2
-        names = @index[location]
-        if names.is_a?(Numeric)
-          row = @data.map { |vector| vector[location] }
-
-          return Daru::Vector.new(row, index: @vectors, name: coerce_name(location))
-        end
+      if positions.is_a? Numeric
+        return Daru::Vector.new populate_row_for(positions),
+          index: @vectors,
+          name: indexes.first
+      else
+        new_rows = @data.map { |vec| vec[*indexes] }
+        return Daru::DataFrame.new new_rows,
+          index: @index.subset(*indexes),
+          order: @vectors
```

Comment:
If I'm not missing something, this way there would be less translations and checks:
```ruby
new_rows = @data.map { |vec| vec.at(*positions) }
return Daru::DataFrame.new new_rows,
  index: @index.at(*positions),
  order: @vectors
```